pkg/segment/query/segquery.go

```diff
@@ -516,7 +516,7 @@ func getSortedQSRResult(aggs *structs.QueryAggregators, allQSRs []*QuerySegmentR
 		} else {
 			// index 0 should have the earliest time
 			sort.Slice(allQSRs, func(i, j int) bool {
-				return allQSRs[i].segKeyTsRange.StartEpochMs > allQSRs[j].segKeyTsRange.StartEpochMs
+				return allQSRs[i].segKeyTsRange.EndEpochMs > allQSRs[j].segKeyTsRange.EndEpochMs
```

Comment:
Is this going to break other things? If other code relies on what it was, then you might want to do this custom sort inside searcher.go (also, you might want to sort in searcher.go anyways to handle `recentFirst` and `recentLast` sorting modes)

Reply:
Currently, all tests seemed to have passed, but I think we should update it just so that we don't assume the current code and ground truth and build tests which might be incorrect.

Reply:
I'm not quite following. I think with your most recent commits, you don't rely on this change anymore; do you think there's a reason to keep this change? I'm not sure if by "current code" you mean the the unmodified version of this or the modified version

Reply:
I meant that the existing code (unmodified version) is incorrect and if someone uses this method, this can cause issues. Although it is not being used right now with my change. I thought since, its a public method, it might be better to correct it here as well.